### PR TITLE
Validation improvement for array schema items

### PR DESCRIPTION
**Pull request by jemerald**
### Description
Fixes: #1767 

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jemerald** [comment]
@shockey the one test that failed on TravisCI does not fail on my local run... This same thing happen with my previous PR as well... These async tests seems to be rather flaky....

File: test/plugins/validate-semantic/2and3/schemas.js [hunk 1]
```diff
@@ -88,6 +88,35 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
...
+      const spec = {
+        swagger: "2.0",
+        "paths": {
+          "/pets": {
+            "get": {
+              "responses": {
+                "200": {
+                  schema: {
+                    name: "myParam",
+                    in: "query",
```
Comment (hkosova) [inline test/plugins/validate-semantic/2and3/schemas.js]: Please remove `name` and `in` - responses do not use these keywords.

Comment (jemerald) [reply test/plugins/validate-semantic/2and3/schemas.js]: Removed

**Comment by jemerald** [comment]
Duplicate of https://github.com/swagger-api/swagger-editor/pull/1769

**Comment by shockey** [comment]
@jemerald, my apologies, I didn't see your PR before I created my own 😬 

You're right, the async tests are a bit flaky. The issue is not trivial to fix, unfortunately - but it is on our radar.

**Pull request closed without merging**